### PR TITLE
Add a theme description to readme and stylesheet

### LINKED ISSUE #119: Add a theme description
**Describe the bug**
The theme is missing a description, this needs to be added to 
https://github.com/WordPress/twentytwentyone/blob/trunk/assets/sass/01-settings/file-header.scss#L6
and the readme.txt file.

### PR DESCRIPTION
**Pull request by pattonwebz**
## Summary
Adds a description for the theme in all the relevent places. There description in this PR I feel is a good one to use based on discussions on the linked issue. There may still need to be minor tweaks to it before it is the _final, final_ version but I think what is in this PR is a good balance between all the suggestions.

Props to: @francescamarano, @marybaum, @melchoyce from the original thread for their suggestions.

Closes #119 

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+Contributors: wordpressdotorg
+Requires at least: 5.3
+Tested up to: 5.5
+Requires PHP: 5.6
+License: GPLv2 or later
+License URI: http://www.gnu.org/licenses/gpl-2.0.html
 
 ## Description
 
+Twenty Twenty-One is a blank canvas for your ideas - it makes the block editor your best brush.
```
Comment (melchoyce) [inline README.md]: This should be an em dash —

File: README.md [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+
+With new block patterns, which allow you to create a beautiful layout in a matter of seconds, this theme’s soft colors and eye-catching—yet timeless—design will let your work shine.
```
Comment (melchoyce) [inline README.md]: The dashes here are also incorrect — it should be:

> eye-catching — yet timeless — design

**Review by melchoyce — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (melchoyce) [reply README.md]: Actually, let's drop the em dash entirely and go with:

> Twenty Twenty-One is a blank canvas for your ideas and it makes the block editor your best brush.

File: README.md (hunk 2 above)
Comment (melchoyce) [reply README.md]: Looks like the issue here is actually GitHub's monospace font, copying it into the editor shows the em dashes correctly.

File: README.md (hunk 1 above)
Comment (pattonwebz) [reply README.md]: Made this change

File: README.md (hunk 2 above)
Comment (pattonwebz) [reply README.md]: Swapped, good catch on this :+1:

**Review by melchoyce — APPROVED** [review]
Looks good 👍 Thanks!

**Pull request merged**